packages/clerk-js/src/core/fraudProtection.ts

```diff
@@ -1,12 +1,16 @@
 import { CaptchaChallenge } from '../utils/captcha/CaptchaChallenge';
 import type { Clerk } from './resources/internal';
-import { Client, isClerkAPIResponseError } from './resources/internal';
+import { ClerkRuntimeError, Client, isClerkAPIResponseError } from './resources/internal';
 
 export class FraudProtection {
   private static instance: FraudProtection;
 
   private inflightException: Promise<unknown> | null = null;
 
+  private captchaRetryCount = 0;
+  private readonly MAX_RETRY_ATTEMPTS = 3;
+  private hasReachedDeadEnd = false;
```

Comment:
```suggestion
  private captchaAttemptsExceeded = false;
```

but i'm curious why we need an additional variable to track this state.

Reply:
You are right, removed. It's a leftover from extra code I had initially.